File: CrackingTheCodingInterview/RightSideOfTree.java
public class RightSideOfTree {
    public static void main(String[] args) {
        
    }
    public List<Integer> rightSideView(TreeNode root) {
        if (root == null) {
            return new ArrayList<Integer>();
        }
        List<Integer> res = new ArrayList<>();
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        while (queue.size() != 0) {
            List<Integer> currentLevelValues = new ArrayList<>();
            int count = 0;
            int length = queue.size();
            while (count < length) {
                TreeNode currentNode = queue.remove();
                currentLevelValues.add(currentNode.val);
                if (currentNode.left != null) {
                    queue.add(currentNode.left);
                }
                if (currentNode.right != null) {
                    queue.add(currentNode.right);
                }
                count++;
            }
            res.add(currentLevelValues.remove(currentLevelValues.size()-1));
        }
        return res;
    }
}

File: CrackingTheCodingInterview/RightSideOfTree.js
const rightSideView = function(root) {
    if (!root) {
        return [];
    }
    let res = [];
    const queue = [root];
    while (root.length) {
        const currentLevelValues = [];
        let length = queue.length;
        let count = 0;
        while (count < length) {
            const currNode = queue.shift();
            currentLevelValues.push(currNode.val);
            if(currNode.left) {
                queue.push(currNode.left)
            }
              
            if(currNode.right) {
                queue.push(currNode.right)
            }
            count++; 
        }
        res.push(currentLevelValues.pop());
    }
    return res;
}